### PR TITLE
Clarify on subscribe rejection reason.

### PR DESCRIPTION
**Pull request by mvalipour**

### CODE REVIEW

File: routemaster/client/connection.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def subscribe(options)
           end
 
           unless response.success?
-            raise 'subscribe rejected'
+            raise "subscribe rejected (#{response.status})"
```
Comment (rxbynerd) [inline routemaster/client/connection.rb]: maybe we can do a proper exception class here, so it's tracked away from your garden variety `RuntimeError`'s in New Relic?

Comment (mvalipour) [reply routemaster/client/connection.rb]: Good idea. I was mainly replicating the existing pattern but totally happy to do that.

**Comment by mvalipour** [comment]
@rxbynerd @mezis updated.

**Comment by mezis** [comment]
Thanks @mvalipour !

**Pull request merged**